.github/workflows/ci-pipeline.yml

```
@@ -7,10 +7,57 @@ on:
   pull_request:
 
 jobs:
-  build:
+  Test_Build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - run: docker build --target test --tag my-test-image .
-    - run: docker run --env-file .env.test my-test-image
+
+    - name: build test image
+      run: docker build --target test --tag my-test-image .
+
+    - name: Run tests
+      run: docker run --env-file .env.test my-test-image
+
+
+
+  Production_Build:
+    name: Build and Publish
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest    
+    needs: Test_Build
+
+    steps:
+      - uses: actions/checkout@v2
+        
+      - name: Build Production Image
+        run: docker build --target production --tag markrobertdunn/todo-app:$GITHUB_SHA .
```

You might already have hit with M9 but we'll want Azure to have access to a stable tag that updates when a new version is released, so pushing commit-versioned images is definitely good but it would be nice to also push a `latest` version. We can do that with one image by tagging it twice, but I believe you will have to push it twice (although the latter should be heavily cached!)
```suggestion
        run: docker build --target production --tag markrobertdunn/todo-app:$GITHUB_SHA --tag markrobertdunn/todo-app .
```